### PR TITLE
MB-4236 Add code for UpdateMTOAgentHandler

### PR DESCRIPTION
**Pull request by sandy-wright**
## Description

This PR fleshes out `UpdateMTOAgentHandler` with a call to the new service object and adds tests for the handler. It also updates the `updateMTOAgent` endpoint description in the `prime.yaml` and fixes a couple of small bugs (a copy pasta bug in `UpdateMTOShipmentAddressHandler` and some bad fake phone number values).

## Setup

Set up your dev environment for testing:

```sh
make db_dev_e2e_populate && make server_run
```

Use the instructions [here](https://github.com/transcom/prime_api_deliverable/wiki/Using-Postman) to set up Postman (or a comparable REST API client) for testing the Prime API.

Once Postman is set up, call the `fetchMTOUpdates` endpoint (`/move-task-orders`) to get a list of all Prime-available moves and their related info. Find any MTO shipment that has at least one MTO agent connected to it. Grab the ID of the MTO agent, the ID of the MTO shipment, and the eTag of the agent. 

_Example output:_
```json
[
    {
        "availableToPrimeAt": "2020-09-17T16:46:45.848Z",
        "createdAt": "2020-09-17T16:46:45.928Z",
        "eTag": "MjAyMC0wOS0xN1QxNjo0Njo0NS45MjgyMVo=",
        "id": "5d4b25bb-eb04-4c03-9a81-ee0398cb779e",
        "isCanceled": false,
        "moveOrder": { ... },
        "moveOrderID": "6fca843a-a87e-4752-b454-0fac67aa4988",
        "mtoShipments": [
            {
                "actualPickupDate": "2020-03-16",
                "agents": [
                    {
                        "agentType": "RELEASING_AGENT",
                        "createdAt": "2020-09-17T16:46:46.576Z",
                        "eTag": "MjAyMC0wOS0xN1QxNjo0Njo0Ni41NzY3MjNa", *
                        "email": "test@test.email.com",
                        "firstName": "Test",
                        "id": "d73cc488-d5a1-4c9c-bea3-8b02d9bd0dea", *
                        "lastName": "Agent",
                        "mtoShipmentID": "475579d5-aaa4-4755-8c43-c510381ff9b5", *
                        "updatedAt": "2020-09-17T16:46:46.576Z"
                    }
                ],
               ...
            }
        ],
        "paymentRequests": [ ... ],
        ...
    },
    ...
]
```

Pull up the `updateMTOAgent` endpoint and enter the two IDs as the `mtoShipmentID` and `agentID` url parameters. Enter the `eTag` value as the `If-Match` header value. For the body, fill in any fields you'd like to try updating. 

_Example body (all other fields should be unchanged):_
```json
{
  "firstName": "Testly",
  "phone": "245-555-0091"
}
```

Keep testing until you're satisfied! Some scenarios to try:

* Try to clear out fields, for example: `"firstName": ""`
* Try to mess with the ID values (both `agentID` and `mtoShipmentID`)
* Try to pass in extra fields or fields that shouldn't be updated
* Try to save incorrectly formatted phone numbers or email addresses
* Try to update a non Prime-available MTO Agent
  * This will be tricky unless you have the GHC API set up for testing (hard) or have the database set up with a viewer that makes it easy to access/update (not so bad)
  * If you have a DB viewer, you can use `fetchMTOUpdates` to grab the `eTag` for an agent, and then go into the database and change the `availableToPrimeAt` value for that move to `null`

## Code Review Verification Steps

* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4236) for this change.
* [Prime API Documentation](https://github.com/transcom/prime_api_deliverable/wiki)

### CODE REVIEW

File: pkg/handlers/primeapi/mto_agent_test.go [hunk 1]
```diff
@@ -5,41 +5,202 @@ import (
...
 
 func (suite *HandlerSuite) TestUpdateMTOAgentHandler() {
-	agent := models.MTOAgent{
-		MTOAgentType: models.MTOAgentReleasing,
+	// Set up db objects
+	agent := testdatagen.MakeMTOAgent(suite.DB(), testdatagen.Assertions{
+		Move: testdatagen.MakeAvailableMove(suite.DB()),
+	})
+
+	firstName := "Test"
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/mto_agent_test.go:29]: Please use our fake names from the spreadsheet here. 
Carol Romilly

File: pkg/handlers/primeapi/mto_agent_test.go [hunk 2]
```diff
@@ -5,41 +5,202 @@ import (
...
+		suite.NoError(params.Body.Validate(strfmt.Default))
+
+		// Run handler and check response
+		response := handler.Handle(params)
+		suite.IsType(&mtoshipmentops.UpdateMTOAgentUnprocessableEntity{}, response)
+
+		// Check error message for the invalid fields
+		agentUnprocessable := response.(*mtoshipmentops.UpdateMTOAgentUnprocessableEntity)
+		_, okFirstName := agentUnprocessable.Payload.InvalidFields["firstName"]
+		//_, okContactInfo := agentUnprocessable.Payload.InvalidFields["contactInfo"]
```
Comment (shimonacarvalho) [inline pkg/handlers/primeapi/mto_agent_test.go:156]: Commented code.

**Review by shimonacarvalho — APPROVED** [review]
Minor changes suggested, looks great to me!

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 2 above)
Comment (sandy-wright) [reply pkg/handlers/primeapi/mto_agent_test.go:156]: Oops, nice catch!

File: pkg/handlers/primeapi/mto_agent_test.go (hunk 1 above)
Comment (sandy-wright) [reply pkg/handlers/primeapi/mto_agent_test.go:29]: Will do!

**Pull request merged**